src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitRepoType.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.DotNet.DarcLib.Helpers;
+
+public enum GitRepoType
+{
+    GitHub,
+    AzureDevOps,
+    Local,
+    None
+}
+
+public static class GitRepoTypeParser
+{
+    public static GitRepoType ParseFromUri(string pathOrUri)
```

Comment:
Shouldn't we pass this around as Uri directly, based on the logic below that yields None if it's not convertible to Uri?

Also, should be simple to cover by tests.

Reply:
I'd have to duplicate the `Uri.TryCreate(pathOrUri, UriKind.Absolute, out Uri parsedUri)` for every call of this then?